pyrefinebio/dataset.py

```diff
@@ -5,6 +5,43 @@
 
 
 class Dataset:
+    """Dataset
+
+    Datasets are collections of experiments and their samples.
+    The dataset can be processed and then downloaded.
```

Comment:
Maybe something a bit more descriptive like "A Dataset needs to be constructed and then processed before it can be downloaded. Requires agreement to the Terms and Conditions via an activated API token."